lib/api/WP_Auth0_Api_Client_Credentials.php

```diff
@@ -77,17 +89,50 @@ protected function handle_response( $method ) {
 
 		$response_body = json_decode( $this->response_body );
 
+		// If we have no access token, something went wrong upstream.
 		if ( empty( $response_body->access_token ) ) {
 			WP_Auth0_ErrorManager::insert_auth0_error( $method, __( 'No access_token returned.', 'wp-auth0' ) );
 			return self::RETURN_ON_FAILURE;
 		}
 
-		try {
-			$this->token_decoded = $this->decode_jwt( $response_body->access_token );
-			return $response_body->access_token;
-		} catch ( Exception $e ) {
-			WP_Auth0_ErrorManager::insert_auth0_error( $method, $e );
-			return self::RETURN_ON_FAILURE;
-		}
+		// Set the transient to expire 1 minute before the token does.
+		$expires_in  = ! empty( $response_body->expires_in ) ? absint( $response_body->expires_in ) : HOUR_IN_SECONDS;
+		$expires_in -= MINUTE_IN_SECONDS;
+
+		// Store the token and scope to check when used.
+		set_transient( self::TOKEN_TRANSIENT_KEY, $response_body->access_token, $expires_in );
+		set_transient( self::SCOPE_TRANSIENT_KEY, $response_body->scope, $expires_in );
+
+		return $response_body->access_token;
+	}
+
+	/**
+	 * Get the stored API token from a transient.
+	 *
+	 * @return string
+	 */
+	public static function get_stored_token() {
+		return get_transient( self::TOKEN_TRANSIENT_KEY );
+	}
+
+	/**
+	 * Delete the stored API token and scope from transients.
+	 */
+	public static function delete_store() {
+		delete_transient( self::TOKEN_TRANSIENT_KEY );
+		delete_transient( self::SCOPE_TRANSIENT_KEY );
+	}
+
+	/**
+	 * Check the stored API token scope for a specific scope.
+	 *
+	 * @param string $scope - Scope to check for.
```

Comment:
maybe clarify here that the scope param is for a single scope vs a space separated string of scopes. This was confusing to me on the initial review because when we build /authorize URLs we have a "scope" parameter that accepts as many scopes as wanted, separated by a space.

Reply:
This is only for the client credentials grant (hence the class name)